test/gtest/multimarginloss.hpp

```diff
@@ -0,0 +1,268 @@
+        }
+        else
+        {
+            std::vector<size_t> in_strides(in_dims.size());
+            in_strides.back() = 1;
+            for(int i = in_dims.size() - 2; i >= 0; --i)
+                in_strides[i] = in_strides[i + 1] * in_dims[i + 1];
+            in_strides[0] *= 2;
+            input = tensor<T>{in_dims, in_strides};
+
+            std::vector<size_t> t_len     = {N};
+            std::vector<size_t> t_strides = {2};
+            target                        = tensor<uint64_t>{t_len, t_strides};
+
+            std::vector<size_t> w_lens    = {C};
+            std::vector<size_t> w_strides = {2};
+            weight                        = tensor<T>{w_lens, w_strides};
+        }
+
+        auto gen_in_value = [](auto...) {
+            return prng::gen_A_to_B<T>(static_cast<T>(-1), static_cast<T>(1));
+        };
+        std::generate(input.begin(), input.end(), gen_in_value);
+        input_dev = handle.Write(input.data);
+
+        for(auto& ptr : target)
+        {
+            ptr = prng::gen_A_to_B<uint64_t>(0, C);
+        }
+        target_dev = handle.Write(target.data);
+
+        auto gen_weight_value = [](auto...) {
+            return prng::gen_A_to_B<T>(static_cast<T>(-1), static_cast<T>(1));
+        };
+        std::generate(weight.begin(), weight.end(), gen_weight_value);
+        weight_dev = handle.Write(weight.data);
+
+        if(reduction_mode == MIOPEN_LOSS_REDUCTION_NONE)
+        {
+            // input is (N, C) -> output is (N)
+            output     = tensor<T>{std::vector<size_t>{N}};
+            ref_output = tensor<T>{std::vector<size_t>{N}};
+        }
+        else
+        {
+            // Tensor with 1 element to store result after reduce
+            output     = tensor<T>{std::vector<size_t>{1}};
+            ref_output = tensor<T>{std::vector<size_t>{1}};
+        }
+        std::fill(output.begin(), output.end(), 0);
+        std::fill(ref_output.begin(), ref_output.end(), 0);
+        output_dev = handle.Write(output.data);
+
+        if(reduction_mode != MIOPEN_LOSS_REDUCTION_NONE)
+        {
+            ws_sizeInBytes = miopen::GetMultiMarginLossForwardWorkspaceSize(handle,
+                                                                            input.desc,
+                                                                            target.desc,
+                                                                            weight.desc,
+                                                                            output.desc,
+                                                                            p,
+                                                                            margin,
+                                                                            reduction_mode);
+            if(ws_sizeInBytes == static_cast<size_t>(-1))
+                GTEST_SKIP() << "Call GetMultiMarginLossForwardWorkspaceSize failed!";
```

Comment:
I guess it is `FAIL()`.

Reply:
fixed it

Reply:
anyway I'm not sure if it should be failed or skipped. I prefer skip because basically there are nothing wrong with the test input or data validation, it just does not have any solvers to handle that cases.

Reply:
Or probably we just removed the solver and got the tests passed.